tasks/connectors/edgescan/lib/kenna_api.rb

```diff
@@ -15,9 +15,8 @@ def initialize(options)
 
         # Converts Edgescan location specifiers and vulnerabilities into Kenna assets and adds them to memory
         def add_assets(edgescan_location_specifiers, edgescan_vulnerabilities)
-          kenna_assets = edgescan_location_specifiers.map(&:to_kenna_asset).append(
-            edgescan_vulnerabilities.reject(&:matching_location_specifier).map(&:to_kenna_asset)
-          ).flatten
+          kenna_assets = edgescan_location_specifiers.map(&:to_kenna_asset).flatten.uniq!
+          kenna_assets.concat(edgescan_vulnerabilities.map(&:to_kenna_asset).uniq! - kenna_assets)
```

Comment:
Could we add some code comments here?

Reply:
Fixed in [80d3673](https://github.com/KennaSecurity/toolkit/pull/423/commits/80d367334c28868936fff425357249bafd6dadd2)